conans/test/functional/command/export/exports_method_test.py

```diff
@@ -0,0 +1,206 @@
+        self.assertIn("pkg/0.1 exports: Copied 1 '.txt' file: file.txt", client.out)
+        self.assertIn("pkg/0.1 export() method: Copied 1 '.md' file: LICENSE.md", client.out)
+
+        layout = client.cache.package_layout(ConanFileReference.loads("pkg/0.1"))
+        exported_files = os.listdir(layout.export())
+        self.assertIn("file.txt", exported_files)
+        self.assertIn("LICENSE.md", exported_files)
+
+    def test_export_method_parent_folder(self):
+        folder = temp_folder()
+        save_files(folder, {"subdir/subdir2/file2.txt": "", "subdir/file1.txt": ""})
+        client = TestClient(current_folder=os.path.join(folder, "recipe"))
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile
+
+            class MethodConan(ConanFile):
+                def export(self):
+                    self.output.info("Executing export() method")
+                    self.copy("*.txt", src="../subdir")
+            """)
+        client.save({"conanfile.py": conanfile})
+        client.run("export . pkg/0.1@")
+        self.assertIn("pkg/0.1 export() method: Copied 2 '.txt' files: file1.txt, file2.txt",
+                      client.out)
+
+        layout = client.cache.package_layout(ConanFileReference.loads("pkg/0.1"))
+        self.assertTrue(os.path.isfile(os.path.join(layout.export(), "file1.txt")))
+        self.assertTrue(os.path.isfile(os.path.join(layout.export(), "subdir2", "file2.txt")))
+
+    def test_export_no_settings_options_method(self):
+        client = TestClient()
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile
+
+            class MethodConan(ConanFile):
+                settings = "os"
+                def export(self):
+                    if self.settings.os == "Windows":
+                        self.copy("LICENSE.md")
+            """)
+        client.save({"conanfile.py": conanfile, "LICENSE.md": "license"})
+        client.run("export . pkg/0.1@", assert_error=True)
+        self.assertIn("ERROR: pkg/0.1: Error in export() method, line 7", client.out)
+
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile
+
+            class MethodConan(ConanFile):
+                default_options = {"myopt": "myval"}
+                def export(self):
+                    if self.default_options["myopt"] == "myval":
+                        self.copy("LICENSE.md")
+            """)
+        client.save({"conanfile.py": conanfile})
+        client.run("export . pkg/0.1@", assert_error=True)
+        self.assertIn("ERROR: pkg/0.1: Error in export() method, line 7", client.out)
+
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile
+
+            class MethodConan(ConanFile):
+                options = {"myopt": ["myval"]}
+                default_options = {"myopt": "myval"}
+                def build(self):
+                    self.output.info("MYOPT: %s" % self.options.myopt)
```

Comment:
This conanfile needs an `export()` function if we want to check that the `options` object is actually restored after running those `export()` methods

Reply:
Done